google/ggrc-core
# GGRC-8268/8146/8182 Issue is not mapped to Assessment if create Issue from unify mapper

# Issue description

Another object is mapped to Assessment instead of Issue if map it from Edit Assessment pop up.
Issue is not mapped to Assessment if create Issue from unify mapper.
Error is appeared when discard changes on Edit popup for issue if create it from unify mapper.

# Steps to test the changes
I.
1. Open any Assessment
2. Edit Assessment
3. Click Map Objects button
4. Select any Issue and Save changes
5. Open Edit pop up again and review mapped objects
**Actual:** Another random object is mapped, mapped issue is not displayed in the Related issues tab
**Expected:** Mapped issue should be displayed in the Related issues tab.

II.
1. Open any Audit
2. Open Assessments tab
3. Click "Create" Assessment button
4. Scroll down and click "Map Objects" button
5. Select Issues object type and click "Create and map new object" link
6. Set all mandatory fields and click Save and Close button
**Actual:** Issue is not mapped to Assessment
**Expected:** Issue is mapped and displayed in Assessment creation pop up

III.
1. Open any Audit
2. Open Assessments tab
3. Click "Create" Assessment button
4. Scroll down and click "Map Objects" button
5. Select Issues object type and click "Create and map new object" link
6. Make any changes and close edit popup
7. Click Discard button
**Actual:** "Uncaught TypeError: Cannot read property 'dispatch' of null" warning floating message is appeared, user is not able to close edit popup
**Expected:** Edit popup is closed after clicking Discard button, no any floating warnings

# Solution description

Add check for 'Issue' type in mapper-utils.js. We should make Relationship request with another destination property.
Add event listener in object-mapper.js in case we dispatch MAP_OBJECTS event when map new object to new Assessment.

# Sanity checklist


# PR Review checklist
